app/src/main/java/com/example/mytestproject/ui/weatherData/WeatherViewModelFactory.kt

```diff
@@ -0,0 +1,20 @@
+package com.example.mytestproject.ui.weatherData
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import com.example.mytestproject.data.repository.weatherRepository.WeatherRepositoryImpl
+import java.lang.IllegalArgumentException
+
+@Suppress("UNCHECKED_CAST")
+class WeatherViewModelFactory(
+   private val weatherRepository: WeatherRepositoryImpl
```

Comment:
Instead of WeatherRepositoryImpl we should use WeatherRepository here. That's a point of interfaces. 
Later we can change the implementation class from WeatherRepositoryImpl to another, and we won't need to change anything here.